test/unit/sdk_reporter/test_sdk_reporter.py

```diff
@@ -72,6 +79,110 @@ def test_return_false_when_report_throws_error(self):
+            'agentProfile': ANY,
+            'contentType': 'application/json',
+            'profilingGroupName': lambda_one_click_profiling_group_name
+        }
+        self.client_stubber.add_response('post_agent_profile', {}, expected_params_one_click)
+
+        os.environ.__setitem__('AWS_CODEGURU_PROFILER_GROUP_NAME', lambda_one_click_profiling_group_name)
+        os.environ.__setitem__('HANDLER_ENV_NAME_FOR_CODEGURU', 'test-handler')
+        self.subject.metadata = test_agent_metadata_for_lambda
+        with self.client_stubber:
+            assert self.subject.report(profile) is False
+            assert self.subject.report(profile) is True
+
+    def test_create_pg_not_invoked_in_non_lambda_case(self):
+        expected_params = {
+            'agentProfile': ANY,
+            'contentType': 'application/json',
+            'profilingGroupName': profiling_group_name
+        }
+        self.client_stubber.add_client_error('post_agent_profile',
+                                             service_error_code="ResourceNotFoundException",
+                                             service_message='Simulated ResourceNotFoundException in '
+                                                             'post_agent_profile call',
+                                             expected_params=expected_params)
+
+        expected_params_post_agent_profile = {
+            'agentProfile': ANY,
+            'contentType': 'application/json',
+            'profilingGroupName': profiling_group_name
+        }
+        self.client_stubber.add_response('post_agent_profile', {}, expected_params_post_agent_profile)
+
+        expected_response_create_pg = {
+            'profilingGroup': {
+                'agentOrchestrationConfig': {
+                    'profilingEnabled': True
+                },
+                'arn': 'string',
+                'computePlatform': 'AWSLambda',
+                'createdAt': datetime(2015, 1, 1),
+                'name': 'string',
+                'profilingStatus': {
+                    'latestAgentOrchestratedAt': datetime(2015, 1, 1),
+                    'latestAgentProfileReportedAt': datetime(2015, 1, 1),
+                    'latestAggregatedProfile': {
+                        'period': 'PT5M',
+                        'start': datetime(2015, 1, 1)
+                    }
+                },
+                'tags': {
+                    'string': 'string'
+                },
+                'updatedAt': datetime(2015, 1, 1)
+            }
+        }
+        self.client_stubber.add_response('create_profiling_group', expected_response_create_pg)
+
+        with self.client_stubber:
+            assert self.subject.report(profile) is False
+            assert self.subject.report(profile) is True
+            try:
+                self.client_stubber.assert_no_pending_responses()
+                assert False
+            except AssertionError:
+                assert True
```

Comment:
What is the usecase when the failure of the assertions should not be considered as a failure? I think this is the equivalent of not having any assertions, so if it's not reliable or useful, we can just delete it.

Reply:
I got it, client_stubber.assert_no_pending_responses() should throw exception. What's the usecase? If this is really difficult to test, try to assert something more, like the message or something so we do not hide any failures that we do not want.

Reply:
Ah makes sense. I removed the unnecessary check and just used self.client_stubber.assert_no_pending_responses() like we have in our other test files (ex. test_end_to_end_profiling.py)